language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/GenericSemigroup.hs

```diff
@@ -0,0 +1,21 @@
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Language.PlutusCore.Evaluation.Machine.GenericSemigroup where
+
+import Data.Monoid (Monoid(..))
+import Data.Semigroup (Semigroup(..))
+import GHC.Generics
+-- TODO when this import conflicts, delete this file.
+import Data.Semigroup.Generic
+
+newtype GenericSemigroupMonoid a =
+  GenericSemigroupMonoid { getGenericSemigroupMonoid :: a }
```

Comment:
Probably worth placing somewhere near `PlutusPrelude`.

Reply:
Nah, it should be out as soon as we're on `lts-14`.

Reply:
Add a note to that effect.

Reply:
It's in a different file, saying as soon as it conflicts via name IIRC. You'll see it when it happens, because of duplicate name (I think).

Should I put it in a maintenance file on toplevel?

Reply:
Seems okay. I wouldn't add maintenance file. We already have too many places to record stuff to do. For in-code TODOs I think a `TODO` comment is fine.